### PR TITLE
FWI-1687 - Rename and break out Insights CLI `policy sync`

### PR DESCRIPTION
**Pull request by ivanfetch-fw**
Note: [see the snapshot builds](https://app.circleci.com/pipelines/github/FairwindsOps/insights-cli/200/workflows/5d6cc3c0-69b4-4669-90d4-d6a90d133698/jobs/242/artifacts) for .tar.gz files, if you want to test this CLI locally. Ignore the old version number tagged on the snapshots.

## Checklist
* [x] I have signed the CLA
* [x] I have updated/added any relevant documentation

### What's the goal of this PR?
Improve Insights CLI usability by Refactoring sub-commands and flags, clarifying log and help output, not requiring the Insights organization and token for commands like `help`, and adding an option to suppress graphical-tree characters in output.

### What changes did you make?

* Rename `policy sync|list` sub-commands to `push opa|rules|all` and `list opa|rules|all` respectively,

* `push all` reports partial success if it's unable to push one of OPA policies or rules.

* Add respective `--push-opa-subdirectory` and `--push-rules-subdirectory` flags to `push all|opa|rules` commands to allow overriding our default of `checks` and `rules`.

* Update log and help output to refer to checks as OPA policies, and
rules as automation rules.

* Provide an option to suppress the graphical tree characters in output,
such as `list all`, to make output more readable by screen-readers or
consumable by tools like grep.

* Do not require Insights token and organization to be specified to
display help, or when running commands that do not require access to
Insights.

### Sample Output

* Partial success pushing all:
```
$ ./cli push all -d dir_with_policies_but_not_rules_subdir
INFO[0000] Pushing OPA policies and automation rules to Insights. 
INFO[0000] Updating v1 OPA policy: test                 
INFO[0000] Updating v2 OPA policy: test2                
ERRO[0000] Unable to push rules: stat dir_with_policies_but_not_rules_subdir/rules: no such file or directory 
FATA[0000] Push partially failed.                       
```

* Pushing OPA policies with the `--delete` flag to remove policies from Insights that are not in my directory. The V2 policy `test2` remains, and other V1 policies and their instances are deleted.
```
$ ./cli push opa --delete -d my_dir
INFO[0000] Deleting instance: deployments for OPA policy test 
INFO[0000] Deleting instance: deployments2 for OPA policy test 
INFO[0000] Deleting OPA policy: test2.5                 
INFO[0000] Deleting OPA policy: test                    
INFO[0000] Updating v2 OPA policy: test2                
INFO[0000] Push succeeded.                              
```

[Internal Ticket FWI-1687](https://fairwinds.myjetbrains.com/youtrack/issue/FWI-1687)

### CODE REVIEW

File: cmd/insights/root.go [hunk 1]
```diff
@@ -68,20 +70,35 @@ func exitWithError(message string, err error) {
...
 	rootCmd.PersistentFlags().StringVarP(&configFile, "config", "c", "./fairwinds-insights.yaml", "Configuration file")
 	rootCmd.PersistentFlags().StringVarP(&organization, "organization", "", "", "Fairwinds Insights Organization name")
+	rootCmd.PersistentFlags().BoolVarP(&noDecoration, "no-decoration", "", false, "Do not include decorative characters in output, such as tree visualization.")
 }
 
 func preRun(cmd *cobra.Command, args []string) {
+	if noDecoration {
+		treeprint.EdgeTypeLink = ""
+		treeprint.EdgeTypeMid = ""
+		treeprint.EdgeTypeEnd = ""
```
Comment (rbren) [inline cmd/insights/root.go:93]: I think `Mid` and `End` should probably be `"  "` (two or more spaces)

File: cmd/insights/push_all.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	"github.com/fairwindsops/insights-cli/pkg/opa"
+	"github.com/fairwindsops/insights-cli/pkg/rules"
+)
+
+func init() {
+	// This flag sets a variable defined in the parent `push` command.
+	pushAllCmd.PersistentFlags().StringVarP(&pushOPASubDir, "push-opa-subdirectory", "", "checks", "Sub-directory within push-directory, to contain OPA policies.")
```
Comment (rbren) [inline cmd/insights/push_all.go:29]: Let's make this default `opa`. We don't use `checks` anywhere else

File: cmd/insights/root.go (hunk 1 above)
Comment (ivanfetch-fw) [reply cmd/insights/root.go:93]: I've updated this, here's what the output looks like now, is this what you imagine?

```
$ ./cli list all --no-decoration
.
   opa
       test2 (v2)
   rules
       Assign API Action Items
```

Comment (ivanfetch-fw) [reply cmd/insights/root.go]: Robert [replied looks good to me](https://fairwindsops.slack.com/archives/CNQBGRGFN/p1649255121080529?thread_ts=1649248209.397319&cid=CNQBGRGFN).

File: pkg/opa/opa_calls.go [hunk 3]
```diff
@@ -38,7 +39,7 @@ const opaInstanceURLFormat = opaCheckInstancesURLFormat + "/%s"
 // GetChecks queries Fairwinds Insights to retrieve all of the Checks for an organization
 func GetChecks(org, token, hostName string) ([]opaPlugin.OPACustomCheck, error) {
 	url := fmt.Sprintf(opaURLFormat, hostName, org)
-	logrus.Infof("OPA URL: %s", url)
+	logrus.Debugf("OPA URL: %s", url)
```
Comment (makoscafee) [inline pkg/opa/opa_calls.go:42]: any reason for changing this from `infof`?

**Review by makoscafee — APPROVED** [review]
LGTM. just had question on areas you changed from info to debug level log? i would assume debug level is only in development level.

File: pkg/opa/opa_calls.go (hunk 3 above)
Comment (ivanfetch-fw) [reply pkg/opa/opa_calls.go:42]: The thought in changing the log level here is that a user will not want to see the Insights URL for typical usage, however if they want to verify the URL they can enable debug logging (with the `--log-level debug` flag). Changing this from info to debug  helps to clean up the output.

**Pull request merged**